app/actions/remote/systems.ts

```diff
@@ -16,36 +15,89 @@ export type ConfigAndLicenseRequest = {
+    }
+
+    try {
+        const {data: globalPolicy, error: globalPolicyError} = await fetchGlobalDataRetentionPolicy(serverUrl);
+        const {data: teamPolicies, error: teamPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl);
+        const {data: channelPolicies, error: channelPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl, true);
+
+        const hasError = globalPolicyError || teamPoliciesError || channelPoliciesError;
+        if (hasError) {
+            return hasError;
+        }
+
+        return {
+            globalPolicy,
+            teamPolicies: teamPolicies as TeamDataRetentionPolicy[],
+            channelPolicies: channelPolicies as ChannelDataRetentionPolicy[],
+        };
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+};
+
+export const fetchGlobalDataRetentionPolicy = async (serverUrl: string): Promise<{data?: GlobalDataRetentionPolicy; error?: unknown}> => {
     let client;
     try {
         client = NetworkManager.getClient(serverUrl);
     } catch (error) {
         return {error};
     }
 
-    let data = {};
     try {
-        data = await client.getDataRetentionPolicy();
+        const data = await client.getGlobalDataRetentionPolicy();
+        return {data};
     } catch (error) {
         forceLogoutIfNecessary(serverUrl, error as ClientError);
         return {error};
     }
+};
+
+export const fetchAllGranularDataRetentionPolicies = async (
+    serverUrl: string,
+    isChannel = false,
+    page = 0,
+    policies: Array<TeamDataRetentionPolicy | ChannelDataRetentionPolicy> = [],
+): Promise<{data?: Array<TeamDataRetentionPolicy | ChannelDataRetentionPolicy>; error?: unknown}> => {
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
 
     const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-    if (operator) {
-        const systems: IdValue[] = [{
-            id: SYSTEM_IDENTIFIERS.DATA_RETENTION_POLICIES,
-            value: JSON.stringify(data),
-        }];
-
-        operator.handleSystem({systems, prepareRecordsOnly: false}).
-            catch((error) => {
-                logError('An error occurred while saving data retention policies', error);
-            });
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
     }
 
-    return data;
+    const {database} = operator;
+
+    const currentUserId = await getCurrentUserId(database);
+    let data;
+    if (isChannel) {
+        data = await client.getChannelDataRetentionPolicies(currentUserId, page);
+    } else {
+        data = await client.getTeamDataRetentionPolicies(currentUserId, page);
+    }
+    policies.push(...data.policies);
+    if (policies.length < data.total_count) {
+        await fetchAllGranularDataRetentionPolicies(serverUrl, isChannel, page + 1, policies);
```

Comment:
Can you explain a bit more this piece of logic? It is specially weird to me to be asking for the next page instead of the current page, and not in a "loop manner".

Reply:
My biggest worry is that this is a recursion and at the same time is blocking other code from executing until all policies are fetched, perhaps just make sense to remove the policies from blocking the entry or come up with some genius idea so that this does not need recursion or something.

Also all this actions should provide an argument for `fetchOnly` and when is false, store to the database.